### PR TITLE
Updated misc for Department, Course, Section models

### PR DESCRIPTION
**Pull request by gannonprudhomme**
- Adds term to Department and a composite primary key, plus a function for generating the id
- Adds a function to generate course id to Course file
- Fixes some formatting in Course & Section

### CODE REVIEW

File: autoscheduler/scraper/models/course.py [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+    return "".join((dept, course_num, '-', term))
+
+class Course(models.Model):
+    """ Basic course model """
+
     id = models.CharField(max_length=10, primary_key=True) # Could be dept+course_num
     dept = models.CharField(max_length=4, db_index=True) # CSCE
-    course_num = models.CharField(max_length=5, db_index=True) # i.e. 221
-    title = models.CharField(max_length=100) # Course title, i.e. "Data Structres & Algorithms"
+    course_num = models.CharField(max_length=5, db_index=True) # i.e. 314
```
Comment (gannonprudhomme) [inline autoscheduler/scraper/models/course.py]: Data Structures & Algorithms was too long to fit on the same line, so changed to 314/Programming Languages

File: autoscheduler/scraper/models/course.py [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+    title = models.CharField(max_length=100) # Course title, i.e. "Programming Languages"
```
Comment (gannonprudhomme) [inline autoscheduler/scraper/models/course.py]: Data Structures & Algorithms was too long to fit on the same line, so changed to 314/Programming Languages

File: autoscheduler/scraper/models/department.py [hunk 3]
```diff
@@ -1,9 +1,17 @@
 from django.db import models
 
+def generate_department_id(code: str, term: str):
+    """ Generates the department id as {code}-{term} """
+
+    return "".join((code, '-', term))
```
Comment (gannonprudhomme) [inline autoscheduler/scraper/models/department.py]: Super simple function, thought join was the most more readable than doing `{code}-{term]`, could also change to `code + '-' + term`, bit of a nitpick though

File: autoscheduler/scraper/models/course.py [hunk 4]
```diff
@@ -1,17 +1,27 @@
 from django.db import models
 
-class Course(models.Model): 
+def generate_course_id(dept: str, course_num: str, term: str):
+    """ Generates the coures_id in the form of {dept}{course_num}-{term} """
+
+    return "".join((dept, course_num, '-', term))
```
Comment (gannonprudhomme) [inline autoscheduler/scraper/models/course.py]: Super simple function, thought join was the most more readable than doing `f'{dept}{course_num}-{term]'`, could also change to `dept + course_num + '-' + term`, bit of a nitpick though

File: autoscheduler/scraper/models/department.py [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
+
 class Department(models.Model):
     """ Data structure that stores the department tag and description """
-    code = models.CharField(max_length=4, primary_key=True) # Ex: CSCE
+
+    id = models.CharField(max_length=10, primary_key=True) # code + term
+    code = models.CharField(max_length=4) # Ex: CSCE
     description = models.TextField(max_length=100) # Ex: CSCE - Computer Sci & Engr
+    term = models.CharField(max_length=6, db_index=True)
```
Comment (gannonprudhomme) [inline autoscheduler/scraper/models/department.py]: There are different departments for a given semester(for instance, there are 45 more departments in fall 2019 then fall 2013)

File: autoscheduler/scraper/models/course.py [hunk 6]
```diff
@@ -1,17 +1,27 @@
 from django.db import models
 
-class Course(models.Model): 
+def generate_course_id(dept: str, course_num: str, term: str):
+    """ Generates the coures_id in the form of {dept}{course_num}-{term} """
```
Comment (ryan-conn) [inline autoscheduler/scraper/models/course.py]:
```suggestion
    """ Generates the course_id in the form of {dept}{course_num}-{term} """
```

**Comment by gannonprudhomme** [comment]
Requested a review from you Brandon first so not only you know what changes are coming(mainly that I'm adding a term attribute to `Department), but mainly b/c we want to start including everyone in pull requests(which we discussed on Sunday)

**Comment by ryan-conn** [comment]
Looks good, I agree with all these changes

**Pull request merged**